### DISCUSSION TITLE
[Core Protocol] Optimism private mempool vs pending blocks

### DISCUSSION THREAD

**Discussion by anondomini** [ethereum-optimism/developers#430] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I have a question and concerns regarding Optimism mempool and pending blocks. On the one hand optimism has private mempool implemented: https://docs.optimism.io/chain/differences#mempool-rules
but on the other hand there's a possibility to fetch pending block with transactions: https://docs.optimism.io/builders/app-developers/transactions/statuses#pending however, after invoking the call, it seems that the transactions are not really pending and they are already included within the block that the response returned? Am I missing something? Maybe eth_getBlockByNumber with pending parameter doesn't really return pending block with transactions but safe/latest block? Thanks for the help!

**Comment by imtipi** [comment, marked as answer]
pending means not published on L1 yet,but already appear(finished) in L2 blockchain

**Reply by anondomini** [reply]
gotcha, thanks
